lib/tasks/publishing_api.rake

```diff
@@ -102,6 +102,19 @@ namespace :publishing_api do
     Organisation.find_each(&:publish_to_publishing_api)
   end
 
+  desc "Republish an organisation's About page"
+  task :republish_about_page, [:org_slug] => :environment do |_, args|
+    about_us_page = Organisation.find_by!(slug: args[:org_slug]).about_us
+    PublishingApiDocumentRepublishingWorker.new.perform(about_us_page.document_id)
+  end
+
+  desc "Republish all About pages"
+  task republish_all_about_pages: :environment do
+    about_us_pages = Organisation.all.map(&:about_us).compact
+    about_us_pages.map(&:document_id)
+      .each(&PublishingApiDocumentRepublishingWorker.method(:perform_async))
```

Comment:
Cool use of `each`, didn't know you could use it this way!

Reply:
Should we do this synchronously, so we know if all the jobs completed successfully?

Reply:
Also, I notice the implementation of this task is different to [(re)publishing for organisations](https://github.com/alphagov/whitehall/blob/7b5c5a086b89cb62ffba62b152a0a8dcfc10c8e6/app/models/about_page.rb). The technique we're using here seems better, since it should also update the draft stack. Are we confident we can republish the new org pages on the draft stack too?

Reply:
Hi Ben - I have just re-tested this on Integration and running `publishing_api:republish_organisation["government-digital-service"]` updated both the live and draft stacks.

Reply:
> Should we do this synchronously, so we know if all the jobs completed successfully?

Performing it asynchronously follows the convention of the other rake tasks. I've now added some logging, which helps to provide at least some visibility of what the rake task is doing - again this is a convention as per other rake tasks. See `patch_published_item_links` for a similar example.

Reply:
We seem to have a mixture of sync and async. For example, the [task to republish all orgs](https://github.com/alphagov/whitehall/blob/795d103153f516ee665f0dc9008025be422cfc1b/lib/tasks/publishing_api.rake#L102) appears to be synchronous. How will we verify the about pages were all republished successfully? The synchronous approach should make this clear - worth noting Whitehall has historical problems with the failing async jobs going unnoticed.

Reply:
Have now changed to synchronous 👍